src/main/scala/com/mwz/sonar/scala/ScalaPlugin.scala

```diff
@@ -46,16 +46,20 @@ object Scala {
   private val FileSuffixesPropertyKey = "sonar.scala.file.suffixes"
   private val DefaultFileSuffixes = Array(".scala")
   private val ScalaVersionPropertyKey = "sonar.scala.version"
-  private val DefaultScalaVersion = ScalaVersions.Scala_2_11.toString()
+  private val DefaultScalaVersion = ScalaVersions.Scala_2_11
   private val SourcesPropertyKey = "sonar.sources"
   private val DefaultSourcesFolder = "src/main/scala"
 
-  def getScalaVersion(settings: Configuration): String =
-    settings.get(ScalaVersionPropertyKey).toOption.getOrElse(DefaultScalaVersion)
+  def getScalaVersion(settings: Configuration): ScalaVersion =
+    settings
+      .get(ScalaVersionPropertyKey)
+      .toOption
+      .flatMap(ScalaVersion.parse)
+      .getOrElse(DefaultScalaVersion)
```

Comment:
It might also be worth to distinguish a missing version from an incorrect/unsupported version with the latter possibly returning an error and terminating the analysis, but I'll leave it to you to decide whether this should be a part of this PR :+1:

Reply:
I don't know if it is possible to stop the analysis if the property is bad configured. well, here I'm talking about all the analysis process, not only the scoverage sensor.
Also I don't know if that would be better instead of just returning a default value.

In any case, in my opinion, it is outside the scope of this PR.

But what I consider worth would be to **log a _warning_** if the default version is returned. I don't know if you would like to add such logging statement inside the `getOrElse` method, or if you would prefer to separate the logging from the logic (like in `ScoverageSensor`).

Reply:
Yes, adding a separate warning message would be definitely useful 👍